homeassistant/components/go2rtc/client.py

```diff
@@ -0,0 +1,149 @@
+"""The go2rtc component."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+import logging
+from typing import TYPE_CHECKING
+
+from go2rtc_client.ws import (
+    Go2RtcWsClient,
+    ReceiveMessages,
+    WebRTCAnswer,
+    WebRTCCandidate,
+    WebRTCOffer,
+    WsError,
+)
+from webrtc_models import RTCIceCandidateInit
+
+from homeassistant.components.camera.webrtc import (
+    WebRTCAnswer as HAWebRTCAnswer,
+    WebRTCCandidate as HAWebRTCCandidate,
+    WebRTCError,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+
+from . import Go2RtcData, async_is_supported
+
+if TYPE_CHECKING:
+    from homeassistant.components.camera import Camera, WebRTCMessage, WebRTCSendMessage
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Go2RtcClient:
+    """Go2rtc client."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        data: Go2RtcData,
+        camera: Camera,
+        client_remove_fn: Callable[[], None],
+    ) -> None:
+        """Initialize the Go2rtc client."""
+        self._hass = hass
+        data.ha_clients.append(self)
+        self._data = data
+        self._camera = camera
```

Comment:
Why do we pass and store the camera? It's not used.

I think we should avoid storing the entity instance.